code/modules/clothing/gloves/_gloves.dm

```diff
@@ -41,12 +41,15 @@
 		var/mob/M = loc
 		M.update_worn_gloves()
 
-/obj/item/clothing/gloves/wirecutter_act(mob/living/user, obj/item/I)
-	. = ..()
-	if(!cut_type)
-		return
-	if(icon_state != initial(icon_state))
-		return // We don't want to cut dyed gloves.
-	new cut_type(drop_location())
-	qdel(src)
-	return TRUE
+/obj/item/clothing/gloves/attackby(obj/item/I, mob/user, params)
+	if(do_after(user, 3 SECONDS, target=src))
+		if(I.tool_behaviour == TOOL_WIRECUTTER || I.get_sharpness())
```

Comment:
You'll want to turn these checks into their own proc that you run before and after the do_after.

Otherwise attacking even cut gloves with any item will trigger a 3 second delay before eventually not doing anything.

Reply:
I couldn't figure out how to do that, but i did figure out another way of stopping the delay from showing up when it shouldn't.

Reply:
What you did is incorrect, because you need to perform the checks *after* the do_after as well.

Simply:

```dm
/obj/item/clothing/gloves/proc/can_cut_with(obj/item/tool)
    if (!check 1)
        return FALSE
    // etc
    return TRUE
```

Then call this before and after the do_after.

Reply:
Forgive me if this question sounds stupid, but why would you need to do the checks after the do_after?

If the checks pass the first time, i can't really think of a reason why you would need to do them a second time, unless there's some bug i don't know about.

Reply:
Two people cutting the same pair of gloves is the most obvious to me.